hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieTimeline.java

```diff
@@ -335,7 +336,12 @@ public interface HoodieTimeline extends Serializable {
   /**
    * @return Get the stream of completed instants
    */
-  Stream<HoodieInstant> getInstants();
+  Stream<HoodieInstant> getInstantsAsStream();
+
+  /**
+   * @return Get tht list of instants
+   */
+  List<HoodieInstant> getInstants();
```

Comment:
Should be ok to change. I have checked the usages of HoodieTimeline in other query engines Trino/Presto. We are good.

Reply:
Thanks a lot @codope